internal/oci/uvm.go

```diff
@@ -352,9 +352,21 @@ func SpecToUVMCreateOpts(ctx context.Context, s *specs.Spec, id, owner string) (
 // UpdateSpecFromOptions sets extra annotations on the OCI spec based on the
 // `opts` struct.
 func UpdateSpecFromOptions(s specs.Spec, opts *runhcsopts.Options) specs.Spec {
-	if opts != nil && opts.BootFilesRootPath != "" {
+	if opts == nil {
+		return s
+	}
+
+	if opts.BootFilesRootPath != "" {
 		s.Annotations[annotationBootFilesRootPath] = opts.BootFilesRootPath
 	}
 
+	if _, ok := s.Annotations[annotationProcessorCount]; !ok && opts.VmProcessorCount != 0 {
```

Comment:
I like what you did here. Lets give the annotation preference as the override for boot files path as well.